### PR TITLE
Enable native code on aarch64-*-openbsd*

### PR DESCRIPTION
**Pull request by madroach**
I enabled aarch64 native code on OpenBSD. I am able to build OCaml and all our OCaml ports.

But still I get a lot of errors like this:
```
 ... testing 'gcwords.ml' with 1 (native) => failed (Files /usr/obj/pobj/ocaml-4.11.2/ocaml-4.11.2/testsuite/_ocamltest/tests/misc/gcwords/ocamlopt.byte/gcword
s.opt and /usr/obj/pobj/ocaml-4.11.2/ocaml-4.11.2/testsuite/_ocamltest/tests/misc/gcwords/ocamlopt.opt/gcwords.opt are different)
```

comparing the disassembly yields this:

``` diff
--- /tmp/gcwords_byte.opt.dis   Mon Dec 20 12:26:17 2021
+++ /tmp/gcwords_opt.opt.dis    Mon Dec 20 12:26:01 2021
@@ -1,5 +1,5 @@

-ocamlopt.byte/gcwords.opt:     file format elf64-littleaarch64
+ocamlopt.opt/gcwords.opt:      file format elf64-littleaarch64


 Disassembly of section .interp:
@@ -386284,8 +386284,8 @@
        0: 00 63 61 6d          ldp     d0, d24, [x24, #-496]
        4: 6c 73 74 61          <unknown>
        8: 72 74 75 70          adr     x18, #962191
-       c: 30 66 63 61          <unknown>
-      10: 37 32 2e 73          <unknown>
+       c: 63 31 63 64          <unknown>
+      10: 63 65 2e 73          <unknown>
       14: 00 24 64 2e          uhsub   v0.4h, v0.4h, v4.4h
       18: 31 00 24 64          <unknown>
       1c: 2e 32 00 24          <unknown>
@@ -386299,8 +386299,8 @@
       3c: 33 00 24 78          <unknown>
       40: 2e 31 00 63          <unknown>
       44: 61 6d 6c 61          <unknown>
-      48: 73 6d 63 34          cbz     w19, 0xc6df4 <.got.plt+0x81cc>
-      4c: 30 37 32 35          cbnz    w16, 0x64730 <camlStdlib__char__escaped_88+0xd8>
+      48: 73 6d 63 65          <unknown>
+      4c: 34 32 64 63          <unknown>
       50: 2e 73 00 63          <unknown>
       54: 61 6d 6c 61          <unknown>
       58: 73 6d 31 35          cbnz    w19, 0x62e04 <camlStdlib__input_line_318+0x4>
```

both differences are in section

```
Disassembly of section .strtab:

0000000000000000 <.strtab>:
```

I'm not familiar with ELF, but I suspect strtab is not supposed to contain instructions anyway, so comparing disassembly makes little sense.

Now, are these failing tests relevant?

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -374,6 +374,9 @@ OCaml 4.14.0
 
 ### Build system:
 
+- Build native-code compilers on OpenBSD
```
Comment (avsm) [inline Changes]:
```suggestion
- Build native-code compilers on OpenBSD/aarch64
```

**Comment by madroach** [comment]
ok. running the testsuite again the failures about bytecode / nativecode differences vanished. This might be because I uninstalled system OCaml before running the testsuite. Could it be that the testsuite accidentally calls system ocaml compilers instead of the newly built compilers ?

**Comment by tomsmeding** [comment]
For what it's worth, the differing parts of the ELF file in the original post appear to be ASCII strings (also evident from the section name, `strtab`); the differing strings are the following:

```
>>> "".join(chr(x) for x in [0x63, 0x61, 0x6d, 0x6c, 0x73, 0x74, 0x61, 0x72, 0x74, 0x75, 0x70, 0x30, 0x66, 0x63, 0x61, 0x37, 0x32, 0x2e, 0x73])
'camlstartup0fca72.s'
>>> "".join(chr(x) for x in [0x63, 0x61, 0x6d, 0x6c, 0x73, 0x74, 0x61, 0x72, 0x74, 0x75, 0x70, 0x63, 0x31, 0x63, 0x64, 0x63, 0x65, 0x2e, 0x73])
'camlstartupc1cdce.s'
>>> "".join(chr(x) for x in [0x63, 0x61, 0x6d, 0x6c, 0x61, 0x73, 0x6d, 0x63, 0x34, 0x30, 0x37, 0x32, 0x35, 0x2e, 0x73])
'camlasmc40725.s'
>>> "".join(chr(x) for x in [0x63, 0x61, 0x6d, 0x6c, 0x61, 0x73, 0x6d, 0x63, 0x65, 0x34, 0x32, 0x64, 0x63, 0x2e, 0x73])
'camlasmce42dc.s'
```

Now I know nothing about ocaml (stumbled upon this PR randomly), but I'd guess these are names of the assembly files (`.s` files) generated during compilation, and for some reason (error messages?) embedded in the executable. So these differences indeed sound related to the compilation toolchain.

**Comment by madroach** [comment]
I believe this can be merged now.

**Comment by dra27** [comment]
Would you be able to rebase this, please? It does look good to go, yes - the program comparison test (comparing ocamlopt.byte and ocamlopt.opt output) is a slightly contentious step, as it is!

**Comment by madroach** [comment]
… I did rebase. See 9279f6fd650e6ce1d6612da1a5175e3d24077773. But aarch64 support will drop in trunk, ~~so how do I tell  github to target the 4.14 branch instead of trunk?~~
… managed to change destination bracnh to 4.14 :-)

**Comment by avsm** [comment]
ARM64 will hopefully be back in trunk soon with #10972. If you get a chance to combine #10875 and #10972 and let us know if it works that would be helpful @madroach. I'm pulling an rPi out of the locker to install OpenBSD/arm64 on, but won't have time till next week to get it up and running (the install process looks a bit more convoluted than I'd hoped)

**Comment by dra27** [comment]
It's OK - please could the base stay as `trunk` but - assuming @Octachron is happy - it'll get cherry-picked to 4.14

**Comment by dra27** [comment]
Scratch that - sorry! I realise that it's commented out on trunk at the moment, which is awkward! @Octachron - is this OK for 4.14 and I'll cherry-pick back to trunk?

**Comment by dra27** [comment]
I think it might need an actual rebase on to 4.14 itself, though, which might kick CI into action?

**Comment by Octachron** [comment]
The PR number could be added to the `Change`, but yes otherwise a configuration-only change seems fine for 4.14 .

**Review by dra27 — APPROVED** [review]
LGTM, thanks

**Pull request merged**